### PR TITLE
Typeassertion should be checked

### PR DESCRIPTION
**Pull request by majst01**
metal-core sometimes does not see any lldp packets to phone home, this at least makes the code more robust.

### CODE REVIEW

File: pkg/lldp/client.go [hunk 1]
```diff
@@ -69,7 +69,10 @@ func (l *Client) Start(resultChan chan<- DiscoveryResult) {
 					if layer.LayerType() != layers.LayerTypeLinkLayerDiscoveryInfo {
 						continue
 					}
-					info := layer.(*layers.LinkLayerDiscoveryInfo)
+					info, ok := layer.(*layers.LinkLayerDiscoveryInfo)
+					if !ok {
```
Comment (GrigoriyMikhalkin) [inline pkg/lldp/client.go:73]: So `layer.LayerType() != layers.LayerTypeLinkLayerDiscoveryInfo` check doesn't work?

**Comment by mwindower** [comment]
lgtm

File: pkg/lldp/client.go (hunk 1 above)
Comment (majst01) [reply pkg/lldp/client.go:73]: might be, i dont know, but this is a part in the code where this goroutine could fail, at least it makes it better.

Comment (majst01) [reply pkg/lldp/client.go:73]: The comparison above compares layer.LayerType with `LayerTypeLinkLayerDiscoveryInfo`, and later it is TypeAsserted to `LinkLayerDiscoveryInfo` which is obviosly not the same

**Review by GrigoriyMikhalkin — APPROVED** [review]
lgtm

**Pull request merged**